server/src/core/workspace/services/workspace.service.ts

```diff
@@ -81,4 +81,35 @@ export class WorkspaceService {
 
     return workspace;
   }
+
+  async deleteWorkspace(id: string, select: any) {
+    const workspace = await this.findUnique({ where: { id }, select });
+
+    // Get workspace companies
+    const companies = await this.companyService.findMany({
+      where: { workspaceId: id },
+    });
+
+    // get workspace people
+    const people = await this.personService.findMany({
+      where: { workspaceId: id },
+    });
+
+    // get workspace pipelines
+    const pipelines = await this.pipelineService.findMany({
+      where: { workspaceId: id },
+    });
+
+    // get workspace stages
+    const stages = await this.pipelineStageService.findMany({
+      where: { workspaceId: id },
+    });
+
+    // TODO: Determine order of deletion
+    // Perhaps we don't delete immediately but instead schedule for deletion
```

Comment:
Good point, I would say:
- PipelineProgresses
- Companies
- People
- PipelineStages
- Pipeline
- User
should work!

Please make sure that this is working as expected carefully as we don't want to erase the database by accident :p We unfortunately don't have tests on the backend :facepalm:, we are adding them this week

Reply:
Okay, @charlesBochet. Going to be extra careful with this one

Reply:
Below are the relationships derived from the prisma schema
![Dependency Graph](https://github.com/twentyhq/twenty/assets/140154534/b6914cc1-8191-43ca-90ab-93014d805ac5)

Reply:
@gitstart-twenty All the models will be linked to a `Workspace` so in my opinion cascade delete can be acceptable, what do you think about that @charlesBochet ?

Reply:
I think manually checking for relationships and deleting data in other models before the workspace might get out of hand pretty fast. So, cascade sounds like the way to go.

Reply:
LGTM!

Reply:
Hey @charlesBochet  don't believe soft deletes support cascading deletion as per my reading on the topic, we'd have to manually update each associated record in other models to sync the deletion(set their `deletedAt`, field to the current date)

Reply:
Hey @charlesBochet Under what circumstance should someone be able to delete a workspace? At the moment, the rules let anyone delete a workspace?

Reply:
@gitstart-twenty You're right, adding a `softDelete` middleware doesn't handle cascade soft deletion actually 🤔
Let me think a bit about it !

Reply:
@gitstart-twenty I think this library can be a good way to go:
https://github.com/olivierwilkinson/prisma-soft-delete-middleware
What do you think about that @charlesBochet ?
This one handle nested deletion, using another library that is creating nested middleware.
It doesn't have a lot of stars, but the code seems good !

Reply:
Ok, thanks for the input. Let's not use Cascade Delete as we will likely handle them differently. Let's programmatically trigger all the deletions from here.

@gitstart-twenty right now, anybody in the workspace can delete the workspace